src/thor/matrix_action.cc

```diff
@@ -27,33 +27,30 @@ std::string thor_worker_t::matrix(Api& request) {
   adjust_scores(options);
   auto costing = parse_costing(request);
 
-  // Distance scaling (miles or km)
-  double distance_scale = (options.units() == Options::miles) ? kMilePerMeter : kKmPerMeter;
-
   // lambdas to do the real work
   auto costmatrix = [&](const bool has_time) {
-    return costmatrix_.SourceToTarget(*options.mutable_sources(), *options.mutable_targets(), *reader,
-                                      mode_costing, mode, max_matrix_distance.find(costing)->second,
-                                      has_time, options.date_time_type() == Options::invariant);
+    request.mutable_matrix()->set_algorithm(Matrix::CostMatrix);
+    return costmatrix_.SourceToTarget(request, *reader, mode_costing, mode,
+                                      max_matrix_distance.find(costing)->second, has_time,
+                                      options.date_time_type() == Options::invariant);
   };
   auto timedistancematrix = [&]() {
-    return time_distance_matrix_.SourceToTarget(*options.mutable_sources(),
-                                                *options.mutable_targets(), *reader, mode_costing,
-                                                mode, max_matrix_distance.find(costing)->second,
+    request.mutable_matrix()->set_algorithm(Matrix::TimeDistanceMatrix);
+    return time_distance_matrix_.SourceToTarget(request, *reader, mode_costing, mode,
+                                                max_matrix_distance.find(costing)->second,
                                                 options.matrix_locations(),
                                                 options.date_time_type() == Options::invariant);
   };
 
   if (costing == "bikeshare") {
-    const auto& time_distances =
-        time_distance_bss_matrix_.SourceToTarget(options.sources(), options.targets(), *reader,
-                                                 mode_costing, mode,
-                                                 max_matrix_distance.find(costing)->second,
-                                                 options.matrix_locations());
-    return tyr::serializeMatrix(request, time_distances, distance_scale, MatrixType::TimeDist);
+    request.mutable_matrix()->set_algorithm(Matrix::TimeDistanceMatrix);
+    time_distance_bss_matrix_.SourceToTarget(request, *reader, mode_costing, mode,
+                                             max_matrix_distance.find(costing)->second,
+                                             options.matrix_locations());
+    return tyr::serializeMatrix(request);
   }
 
-  MatrixType matrix_type = MatrixType::Cost;
+  Matrix::Algorithm matrix_algo = Matrix::CostMatrix;
```

Comment:
I removed the MatrixType enum we kept around and replaced it with the PBF's one.